src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketServerConfiguration.java

```diff
@@ -64,41 +64,57 @@
     private final String credentialsId;
     private final String id;
     private String baseUrl;
+    private boolean forcesLegacyBuildStatuses;
     private String serverName;
 
     @DataBoundConstructor
     public BitbucketServerConfiguration(
             String adminCredentialsId,
             String baseUrl,
             @Nullable String credentialsId,
-            @Nullable String id) {
+            @Nullable String id,
+            boolean forcesLegacyBuildStatuses) {
         this.adminCredentialsId = requireNonNull(adminCredentialsId);
         this.baseUrl = requireNonNull(baseUrl);
         this.credentialsId = credentialsId;
         this.id = isBlank(id) ? UUID.randomUUID().toString() : id;
+        this.forcesLegacyBuildStatuses = forcesLegacyBuildStatuses;
+    }
+
+    public BitbucketServerConfiguration(
+            String adminCredentialsId,
+            String baseUrl,
+            @Nullable String credentialsId,
+            @Nullable String id) {
+        this(adminCredentialsId, baseUrl, credentialsId, id, false);
+    }
+
+    public String getAdminCredentialsId() {
+        return adminCredentialsId;
     }
 
     /**
-     * For a given item, returns a global credential provider. The credentials are tracked and
-     * this should be the only way to fetch credentials.
+     * Returns the URL location of the server instance
      *
-     * @param item which will use the credential.
-     * @return credential provider
+     * @return the bitbucket server base URL
      */
-    public GlobalCredentialsProvider getGlobalCredentialsProvider(Item item) {
```

Comment:
Alphabetical sorting strikes again :frowning_face:. The relevant addition is the field `forcesLegacyBuildStatuses` and the extra constructor (as having it disabled feels like the "default" setting)